Framework/DataHandling/src/LoadShape.cpp

```diff
@@ -0,0 +1,214 @@
+#include "MantidDataHandling/LoadShape.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidAPI/Sample.h"
+
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/Exception.h"
+
+#include <Poco/File.h>
+
+namespace Mantid {
+namespace DataHandling {
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(LoadShape)
+
+using namespace Kernel;
+using namespace API;
+using namespace Geometry;
+
+void LoadShape::init() {
+  auto wsValidator = boost::make_shared<CompositeValidator>();
+  wsValidator->add<API::InstrumentValidator>();
+
+  // input workspace
+  declareProperty(
+      make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input,
+                                       wsValidator),
+      "The name of the workspace containing the instrument to add the shape");
+
+  // shape file
+  const std::vector<std::string> extensions{".stl"};
+  declareProperty(
+      make_unique<FileProperty>("Filename", "", FileProperty::Load, extensions),
+      "The name of the file containing the shape. ");
```

Comment:
It's not just the name of the file. It's also the path to the file. And that file is an stl file.

Reply:
I'll mention it's a path name to the file. I won't mention it's an STL file here, because this is mentioned elsewhere in the documentation and is subject to change (e.g. OFF file will be added).

Reply:
Ok. if OFF is going to be added to the same algorithm (not another), then this makes sense.

Reply:
If the algorithm were for STL files only, it would have `STL` in its name.